Public/Src/Cache/ContentStore/Library/FileSystem/PassThroughFileSystem.cs

```diff
@@ -445,89 +446,86 @@ private async Task CopyFileInsideSemaphoreAsync(AbsolutePath sourcePath, Absolut
             }
         }
 
-        private Stream OpenInternal(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
+        private Stream TryOpenFile(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
         {
-                if (OperatingSystemHelper.IsUnixOS)
-                {
-                    if (DirectoryExists(path))
-                    {
-                        throw new UnauthorizedAccessException($"Cannot open directory {path} as a file.");
-                    }
+            if (OperatingSystemHelper.IsUnixOS)
+            {
+                return TryOpenFileUnix(path, accessMode, mode, share, options, bufferSize);
+            }
 
-                    if (mode == FileMode.Open && !FileExists(path))
-                    {
-                        return null;
-                    }
+            return TryOpenFileWin(path, accessMode, mode, share, options, bufferSize);
+        }
 
-                    return new FileStream(path.Path, mode, accessMode, share);
-                }
+        private Stream TryOpenFileUnix(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
+        {
+            if (DirectoryExists(path))
+            {
+                throw new UnauthorizedAccessException($"Cannot open directory {path} as a file.");
+            }
 
-                int access = GetDwAccess(accessMode);
-                options = GetOptions(path, options);
+            if (mode == FileMode.Open && !FileExists(path))
+            {
+                return null;
+            }
 
-                SafeFileHandle handle;
+            return new FileStream(path.Path, mode, accessMode, share, bufferSize, options);
+        }
+        
+        private Stream TryOpenFileWin(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)

```

Comment:
Add more comments describing what method is accomplishing and how, which errors are handled (and which not) etc.